src/plugins/options/options.ts

```diff
@@ -1,119 +1,90 @@
-            <label class="option-label" for=${themeOptionId}-uip-dark>Dark</label>
-        </div>`;
-    this.appendChild($theme);
+    this.options.forEach(option => this.append(option));
   }
 
-  protected renderSettingsControl() {
-    const $settings = document.createElement('div');
-    CSSClassUtils.add($settings, 'uip-option');
-    const settingsControlId = randUID();
-    $settings.innerHTML = `
-        <div class="option-item">
-            <input type="checkbox" id=${settingsControlId}-settings-control name=${settingsControlId}-control
-            class="option-checkbox" settings-control='settings' ${this.root?.settings ? 'checked' : ''}>
-            <label class="option-label" for=${settingsControlId}-settings-control>Settings</label>
-        </div>`;
-    this.appendChild($settings);
-  }
-
-  protected renderEditorControl() {
-    const $editor = document.createElement('div');
-    CSSClassUtils.add($editor, 'uip-option');
-    const editorControlId = randUID();
-    $editor.innerHTML = `
-        <div class="option-item">
-            <input type="checkbox" id=${editorControlId}-settings-control name=${editorControlId}-control
-            class="option-checkbox" editor-control='editor' ${this.root?.editor ? 'checked' : ''}>
-            <label class="option-label" for=${editorControlId}-settings-control>Editor</label>
-        </div>`;
-    this.appendChild($editor);
-  }
-
-  protected renderDirection() {
-    const $dir = document.createElement('div');
-    CSSClassUtils.add($dir, 'uip-option dir');
-    const dirOptionId = randUID();
-    $dir.innerHTML = `
-        <div class="option-item">
-            <input type="radio" id=${dirOptionId}-uip-ltr name=${dirOptionId}-dir direction="ltr"
-            class="option-radio-btn" ${this.root?.direction === 'ltr' ? 'checked' : ''}>
-            <label class="option-label" for=${dirOptionId}-uip-ltr>LTR</label>
-        </div>
-        <div class="option-item">
-            <input type="radio" id=${dirOptionId}-uip-rtl name=${dirOptionId}-dir direction="rtl"
-            class="option-radio-btn" ${this.root?.direction === 'rtl' ? 'checked' : ''}>
-            <label class="option-label" for=${dirOptionId}-uip-rtl>RTL</label>
-        </div>`;
-    this.appendChild($dir);
+  @bind
+  protected _onRootConfigChange(e: CustomEvent) {
+    const option = this.options.get(e.detail.attribute);
+    option?.toggleState(e.detail.value !== null);
   }
 
   @bind
-  protected _onOptionChange(e: Event) {
-    const target = e.target as HTMLInputElement;
-
-    const theme = target.getAttribute('theme');
-    const settings = target.getAttribute('settings-control');
-    const editor = target.getAttribute('editor-control');
-    const dir = target.getAttribute('direction');
+  protected onOptionClick(e: Event) {
+    const option = e.target;
+    if (!(option instanceof UIPOption)) return;
+    this.root?.toggleAttribute(option.attribute, option.active);
```

Comment:
If we anyway incidentally have a check of event target here, then i think we can not have multiple event listeners, but do toggling here
```suggestion
    if (!(option instanceof UIPOption)) return;
    option.toggleState();
    this.root?.toggleAttribute(option.attribute, option.active);
```